draft-ietf-quic-transport.md

```diff
@@ -2678,72 +2678,105 @@ transmission efficiency to underfilled packets.
+same applies to the frames that are contained within lost packets.  Instead, the
+different types of information that might be carried in frames is sent again as
+needed.  New frames and packets are used to carry information that is determined
+to have been lost.
+
+* Application data sent in STREAM frames is retransmitted in new STREAM frames
+  unless the endpoint has sent a RST_STREAM for that stream.  Once an endpoint
+  sends a RST_STREAM frame, no further STREAM frames are needed.
+
+* The most recent set of acknowledgments is sent in ACK frames.  An ACK frame
+  SHOULD contain all unacknowledged acknowledgments, as described in
+  {{sending-ack-frames}}.
+
+* Cancellation of stream transmission, as carried in a RST_STREAM frame, is sent
+  until acknowledged or until acknowledgments for all stream data is received
+  from the peer (that is, either the "Reset Recvd" or "Data Recvd" state is
+  reached on the send stream).  The content of a RST_STREAM frame MUST NOT
+  change when it is sent again.
 
-* ACK and PADDING frames MUST NOT be retransmitted.  ACK frames
-  containing updated information will be sent as described in {{frame-ack}}.
+* Similarly, requests to cancel stream transmission, as encoded in a
+  STOP_SENDING frame, is sent until the receive stream enters either a "Data
+  Recvd" or "Reset Recvd" state, see {{solicited-state-transitions}}.
 
-* STOP_SENDING frames MUST be retransmitted until the receive stream enters
-  either a "Data Recvd" or "Reset Recvd" state.  See
-  {{solicited-state-transitions}}.
+* Connection close signals, including those that use CONNECTION_CLOSE and
+  APPLICATION_CLOSE frames, are not sent again when packet loss is detected, but
+  as described in {{termination}}.
 
-* The most recent MAX_STREAM_DATA frame for a stream MUST be retransmitted until
-  the receive stream enters a "Size Known" state. Any previous unacknowledged
-  MAX_STREAM_DATA frame for the same stream SHOULD NOT be retransmitted since a
-  newer MAX_STREAM_DATA frame for a stream obviates the need for delivering
-  older ones. Similarly, the most recent MAX_DATA frame MUST be retransmitted;
-  previous unacknowledged ones SHOULD NOT be retransmitted.
+* The current maximum data offset is sent in MAX_DATA frames until acknowledged.
+  Note that increases to the offset might separately cause new MAX_DATA frames
+  to be sent.  Care is necessary to limit the number of times this frame is
+  sent.
 
-* BLOCKED, STREAM_BLOCKED, and STREAM_ID_BLOCKED frames SHOULD be retransmitted
-  if the sender is still blocked on the same limit.  If the limit has been
-  increased since the frame was originally sent, the frame SHOULD NOT be
-  retransmitted.
+* The current maximum stream data offset is sent in MAX_STREAM_DATA frames until
+  acknowledged or the receive stream enters a "Size Known" state.  Like
+  MAX_DATA, endpoints need to limit the rate at which this frame is sent.
```

Comment:
In this paragraph, and others to follow, do you mean to say that *every packet* contains this frame from when it's first sent till acknowledgment, or only that it's retransmitted when the carrying packet is declared lost? Whichever it is, I think this could stand to be clearer.

Reply:
I have rephrased this completely.  I've changed this so that retransmission only occurs when the *most recently sent* frame is declared lost.  I think that is much neater.

As you say, it was unclear before, and a valid interpretation of the proposed text was "send a frame in every packet", which I agree is a little silly.